trin-history/src/jsonrpc.rs

```diff
@@ -73,54 +79,132 @@ impl HistoryRequestHandler {
+                                Ok(content_key) => {
+                                    // Check whether we have the data locally.
+                                    let local_content: Option<Vec<u8>> =
+                                        match self.network.overlay.store.read().get(&content_key) {
+                                            Ok(Some(data)) => Some(data),
+                                            Ok(None) => None,
+                                            Err(err) => {
+                                                error!(
+                                                    error = %err,
+                                                    content.key = %content_key,
+                                                    "Error checking data store for content",
+                                                );
+                                                None
                                             }
-                                            None => Ok(Value::Null),
+                                        };
+                                    match local_content {
+                                        Some(content) => Ok(json!({
+                                            "content": hex_encode(content),
+                                            "route": [],
+                                        })),
+                                        None => {
+                                            // Request data from network since it's unavailable locally
+                                            let (content, closest_nodes) = self
+                                                .network
+                                                .overlay
+                                                .lookup_content(content_key.clone())
+                                                .await;
+                                            let content = content.unwrap_or_default();
+                                            let closest_nodes = closest_nodes.unwrap_or_default();
+                                            let closest_nodes: Vec<Value> = closest_nodes
+                                                .iter()
+                                                // Skip over enrs that are unable to be looked up
+                                                .filter_map(|node_id| {
+                                                    let content_id = content_key.content_id();
+                                                    // Return None for enrs that cannot be looked up
+                                                    match self
+                                                        .network
+                                                        .overlay
+                                                        .discovery
+                                                        .find_enr(node_id)
```

Comment:
Hmm, good point, but if I'm understanding correctly, this will require a new message type to make this information available via `self.network.overlay` (aka `OverlayProtocol`). Seems out of scope for this pr to me, and imo not the highest priority (but lmk if you feel otherwise). I'd suggest we track this in an issue for now?